language/bytecode-verifier/src/check_duplication.rs

```diff
@@ -22,133 +27,177 @@ pub struct DuplicationChecker<'a> {
 }
 
 impl<'a> DuplicationChecker<'a> {
-    pub fn verify(module: &'a CompiledModule) -> VMResult<()> {
+    pub fn verify_module(module: &'a CompiledModule) -> VMResult<()> {
+        Self::check_identifiers(module.identifiers())?;
```

Comment:
I usually like pattern matching so I don't forget to check anything. It doesn't work here obviously, but its something we could look into in the future. e.g.

```
let CompiledModuleTables { identifiers, module, constant_pool, signatures, module_handles, struct_handles, function_handles, function_instantiations } = module.view_tables();
...
```
It doesn't buy much, and might not be needed here. But I use a pattern like this in the source language compiler in translations, and it tends to really help me not forget things